File: solutions/java/annalyns-infiltration/1/src/main/java/AnnalynsInfiltration.java
class AnnalynsInfiltration {
    boolean knightIsAwake = true;
    boolean archerIsAwake =true;
    boolean prisonerIsAwake = false;
    boolean petDogIsPresent = false;
    public static boolean canFastAttack(boolean knightIsAwake) {
        return knightIsAwake==true?false:true;
    }

    public static boolean canSpy(boolean knightIsAwake, boolean archerIsAwake, boolean prisonerIsAwake) {
        return knightIsAwake==true && archerIsAwake==false && prisonerIsAwake==false? true:knightIsAwake==false && archerIsAwake==true && prisonerIsAwake==false?true:knightIsAwake==false && archerIsAwake==false && prisonerIsAwake==true?true:knightIsAwake==false && archerIsAwake==true && prisonerIsAwake==true?true:knightIsAwake==true && archerIsAwake==false && prisonerIsAwake==true?true:knightIsAwake==true && archerIsAwake==true && prisonerIsAwake==false?true:knightIsAwake==true && archerIsAwake==true && prisonerIsAwake==true?true:false;
    }

    public static boolean canSignalPrisoner(boolean archerIsAwake, boolean prisonerIsAwake) {
        return archerIsAwake==false && prisonerIsAwake==true?true: archerIsAwake==true && prisonerIsAwake==false?false:archerIsAwake==false && prisonerIsAwake==false?false:false;
    }

    public static boolean canFreePrisoner(boolean knightIsAwake, boolean archerIsAwake, boolean prisonerIsAwake, boolean petDogIsPresent) {
          return knightIsAwake==true && archerIsAwake==true && prisonerIsAwake==true && petDogIsPresent==true?false:   knightIsAwake==false && archerIsAwake==true && prisonerIsAwake==true && petDogIsPresent==true?false:  knightIsAwake==false && archerIsAwake==false && prisonerIsAwake==true && petDogIsPresent==true?true:  knightIsAwake==false && archerIsAwake==false && prisonerIsAwake==false && petDogIsPresent==true?true:  knightIsAwake==false && archerIsAwake==false && prisonerIsAwake==true && petDogIsPresent==false?true:  knightIsAwake==true && archerIsAwake==false && prisonerIsAwake==false && petDogIsPresent==true?true: knightIsAwake==true && archerIsAwake==false && prisonerIsAwake==true && petDogIsPresent==true?true:false; 

    }
}

File: solutions/javascript/annalyns-infiltration/1/annalyns-infiltration.js
const knightIsAwake = true;
const archerIsAwake = true;
const prisonerIsAwake =false;
const petDogIsPresent = false; 
export function canExecuteFastAttack(knightIsAwake) {
  return knightIsAwake==true?false:true;
}

export function canSpy(knightIsAwake, archerIsAwake, prisonerIsAwake) {
  return knightIsAwake==false && archerIsAwake==false && prisonerIsAwake==true ? true: knightIsAwake==false && archerIsAwake==false && prisonerIsAwake==false?false:true;
}
export function canSignalPrisoner(archerIsAwake, prisonerIsAwake) {
  return archerIsAwake==false && prisonerIsAwake==true?true:false;
}

export function canFreePrisoner(
  knightIsAwake,
  archerIsAwake,
  prisonerIsAwake,
  petDogIsPresent,
) {
   return knightIsAwake==false && archerIsAwake==true && prisonerIsAwake==false && petDogIsPresent==false? false: knightIsAwake==false && archerIsAwake==false && prisonerIsAwake==false && petDogIsPresent==true?true: knightIsAwake==false && archerIsAwake==false && prisonerIsAwake==true && petDogIsPresent==false ? true: knightIsAwake==false && archerIsAwake==false && prisonerIsAwake==true && petDogIsPresent ==true ? true: knightIsAwake==false && archerIsAwake==true && prisonerIsAwake==false && petDogIsPresent==false?false: knightIsAwake==false && archerIsAwake==false && prisonerIsAwake==false && petDogIsPresent==false?false: knightIsAwake==true && archerIsAwake==false && prisonerIsAwake==false && petDogIsPresent==true?true: knightIsAwake==true && archerIsAwake==false && prisonerIsAwake==true && petDogIsPresent==true?true:false;
}
